agent/src/beerocks/slave/ap_manager_thread.cpp

```diff
@@ -708,6 +737,51 @@ bool ap_manager_thread::handle_cmdu(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_
         LOG(DEBUG) << "CLIENT_DISALLOW: mac = " << sta_mac << ", bssid = " << bssid;
 
         ap_wlan_hal->sta_deny(sta_mac, bssid);
+
+        // Check if validity period is set then add it to the "disallowed client timeouts" list
+        // This list will be polled in after_select()
+        // When validity period is timed out CLIENT_ALLOW message will be sent.
+        if (request->validity_period()) {
+
+            disallowed_client_params_t disallowed_client_params;
+            disallowed_client_params.mac   = request->mac();
+            disallowed_client_params.bssid = request->bssid();
+
+            // check if STA is already disallowed from this bssid
```

Comment:
You'll need to do the same in the handling of the ALLOW case, so better factor this into a function.